packages/lib/response/service.ts

```diff
@@ -80,86 +83,98 @@ const responseSelection = {
   },
 };
 
-export const getResponsesCacheTag = (surveyId: string) => `surveys-${surveyId}-responses`;
-
-export const getResponseCacheTag = (responseId: string) => `responses-${responseId}`;
+const getResponseBySingleUseIdCacheTags = (surveyId: string, singleUseId: string): string =>
+  `survey-${surveyId}-singleuse-${singleUseId}-responses`;
 
 export const getResponsesByPersonId = async (
   personId: string,
   page?: number
-): Promise<Array<TResponse> | null> => {
-  validateInputs([personId, ZId], [page, ZOptionalNumber]);
-
-  try {
-    const responsePrisma = await prisma.response.findMany({
-      where: {
-        personId,
-      },
-      select: responseSelection,
-      take: page ? ITEMS_PER_PAGE : undefined,
-      skip: page ? ITEMS_PER_PAGE * (page - 1) : undefined,
-    });
-
-    if (!responsePrisma) {
-      throw new ResourceNotFoundError("Response from PersonId", personId);
-    }
-
-    let responses: Array<TResponse> = [];
-
-    responsePrisma.forEach((response) => {
-      responses.push({
-        ...response,
-        person: response.person ? transformPrismaPerson(response.person) : null,
-        tags: response.tags.map((tagPrisma: { tag: TTag }) => tagPrisma.tag),
-      });
-    });
-
-    return responses;
-  } catch (error) {
-    if (error instanceof Prisma.PrismaClientKnownRequestError) {
-      throw new DatabaseError("Database operation failed");
+): Promise<Array<TResponse> | null> =>
+  unstable_cache(
+    async () => {
+      validateInputs([personId, ZId], [page, ZOptionalNumber]);
+
+      try {
+        const responsePrisma = await prisma.response.findMany({
+          where: {
+            personId,
+          },
+          select: responseSelection,
+          take: page ? ITEMS_PER_PAGE : undefined,
+          skip: page ? ITEMS_PER_PAGE * (page - 1) : undefined,
+        });
+
+        if (!responsePrisma) {
+          throw new ResourceNotFoundError("Response from PersonId", personId);
+        }
+
+        let responses: Array<TResponse> = [];
+
+        responsePrisma.forEach((response) => {
+          responses.push({
+            ...response,
+            person: response.person ? transformPrismaPerson(response.person) : null,
+            tags: response.tags.map((tagPrisma: { tag: TTag }) => tagPrisma.tag),
+          });
+        });
+
+        return responses;
+      } catch (error) {
+        if (error instanceof Prisma.PrismaClientKnownRequestError) {
+          throw new DatabaseError("Database operation failed");
+        }
+
+        throw error;
+      }
+    },
+    [`getResponses-${personId}`],
```

Comment:
I think we need to include the page: `getResponses-${personId}-${page}`

Reply:
my concern with `page` is that it can be `undefined`.

is that okay?

Thinking about it now, it as long as it comes `undefined` all the time then it would be fine as a key